Unable to reuse a `SetOptions` instance after first command

Hello, I'm writing an application using Redis, and I'm manipulating keys that are known to be volatile/have a TTL at runtime, so I need to save them with an expiry time. I'm executing a lot of commands with these keys, so I have to check, for each command, if the key has a TTL, and if so, do the appropriate command (for example, use `set_ex` instead of `set` method).

# Problem

Then, I looked at the reference and saw the [`SetOptions`](https://docs.rs/redis/latest/redis/struct.SetOptions.html#) type. I thought I could use it with the [`set_options`](https://docs.rs/redis/latest/redis/trait.AsyncCommands.html#method.set_options) method, but because this struct doesn't implement `Clone`/`Copy`, nor the `set_options` method takes a reference to it instead of ownership, it's moved and we can't use the instance anymore. See the example below:

```rust
let my_options = ...;
redis_conn
    .set_options("mykey", "myvalue", my_options)
    .await?;
redis_conn
    .set_options("mykey2", "myvalue2", my_options)
    .await?; //                        ^^^^^^^^^^ we can't reuse my_options here
```

```rust
error[E0382]: use of moved value: `my_options`
 --> records_lib/src/update_mappacks.rs:159:44
  |
1 |     let my_options = ...;
  |         ---------- move occurs because `my_options` has type `SetOptions`, which does not implement the `Copy` trait
2 |     redis_conn
3 |         .set_options("mykey", "myvalue", my_options)
  |                                          ---------- value moved here
...
6 |         .set_options("mykey2", "myvalue2", my_options)
  |                                            ^^^^^^^^^^ value used here after move
```

# Current workaround

The current workaround is to do this:
```rust
Cmd::set("mykey", "myvalue")
    .arg(&my_options)
    .query_async(redis_conn)
    .await?;
Cmd::set("mykey2", "myvalue2")
    .arg(&my_options)
    .query_async(redis_conn)
    .await?;
```
But this is more verbose.

# Solution

Nothing to say more, just need to implement `Clone`/`Copy` for the `SetOptions` type (it doesn't seem to be a heavy type), or make the `set_options` method take a reference to it instead.

Thanks 😁

--- a/redis/src/commands/mod.rs
+++ b/redis/src/commands/mod.rs
@@ -2098,7 +2098,7 @@ impl ToRedisArgs for Direction {
 ///     con.set_options(key, value, opts)
 /// }
 /// ```
-#[derive(Default)]
+#[derive(Clone, Copy, Default)]
 pub struct SetOptions {
     conditional_set: Option<ExistenceCheck>,
     get: bool,

--- a/redis/src/types.rs
+++ b/redis/src/types.rs
@@ -46,6 +46,7 @@ pub enum Expiry {
 }
 
 /// Helper enum that is used to define expiry time for SET command
+#[derive(Clone, Copy)]
 pub enum SetExpiry {
     /// EX seconds -- Set the specified expire time, in seconds.
     EX(usize),
@@ -60,6 +61,7 @@ pub enum SetExpiry {
 }
 
 /// Helper enum that is used to define existence checks
+#[derive(Clone, Copy)]
 pub enum ExistenceCheck {
     /// NX -- Only set the key if it does not already exist.
     NX,

